### PR TITLE
Vite: Add vue-vite framework for Vue2

### PR DESCRIPTION
**Pull request by IanVS**
Issue: #18293

## What I did

This adds a new framework for vue2 and vite, `@storybook/vue-vite`.  It also adds a reproduction, but it sounds like we may need to comment that out for now until we get vue2 playing nicely in CI.  

TODO:
- [X] Figure out how to avoid installing `vue-loader` into Vite projects.  The same goes for `svelte-loader` as well, in svelte-vite projects. (https://github.com/storybookjs/storybook/pull/19263)

## How to test

- Run `yarn bootstrap --build`
- Run `yarn generate-repros-next --template vue2-vite/2.7-js --local-registry`
- `cd ../repros/vue2-vite/2.7-js/after-storybook`
- `yarn storybook`

### CODE REVIEW

**Comment by IanVS** [comment]
I didn't deploy my repro template, which sounds like it might break CI due to the vue2 renderer not working.  So, I'd say this PR isn't ready to merge, and is blocked by https://github.com/storybookjs/storybook/pull/19207.

File: code/lib/cli/src/repro-templates.ts [hunk 1]
```diff
@@ -67,6 +67,20 @@ const vue3ViteTemplates = {
   },
 };
 
+const vue2ViteTemplates = {
+  'vue2-vite/2.7-js': {
+    name: 'Vue2 Vite (vue 2.7 JS)',
+    script:
+      'yarn create vite . --template vanilla && yarn add --dev @vitejs/plugin-vue2 vue-template-compiler vue@2 && echo "import vue2 from \'@vitejs/plugin-vue2\';\n\nexport default {\n\tplugins: [vue2()]\n};" > vite.config.js',
```
Comment (shilman) [inline code/lib/cli/src/repro-templates.ts:74]: @ndelangen WDYT about this? we said we were going to only use community bootstrap tools for this. do we use this as a placeholder for now & then move to some storybook package later?

Comment (IanVS) [reply code/lib/cli/src/repro-templates.ts:74]: The challenge is, there's no generator / bootstrap for using Vue 2 with Vite, as far as I could find.  Vue 2 had its own cli, and it's necessary to create a Vue 2 project and then manually convert it over to Vite.  Or, this approach which required fewer changes, which is to add a vanilla Vite project, and then add Vue2 into it.

**Comment by IanVS** [comment]
I'm going to leave a TODO comment in the code to say that we should create a `npm create` script for vue2 + vite, and we'll move over to that as soon as it's published.

Also, I'll make a PR to remove the `extraPackages` from the vue/svelte generators to remove the loaders, and see what happens.

File: code/lib/cli/src/repro-templates.ts (hunk 1 above)
Comment (ndelangen) [reply code/lib/cli/src/repro-templates.ts:77]: understood! thank you for the explanation, let's do a follow up at some point to extract this into the public domain in some sort of CLI project generator.

**Comment by ndelangen** [comment]
@IanVS Looks like it's not generating in the the directory you want it to:
https://app.circleci.com/pipelines/github/storybookjs/storybook/29387/workflows/f4bae4c8-4f32-4ddc-a5b5-d3d629ebb3a4/jobs/415498
?

**Comment by IanVS** [comment]
I think the issue is that I did not actually build and release the repro into https://github.com/storybookjs/repro-templates-temp.  I was reluctant to do that until https://github.com/storybookjs/storybook/issues/19204 is fixed, since otherwise I guess CI will start to fail everywhere, right @shilman?  I guess this is on-hold until that's figured out.

**Comment by naquiroz** [comment]
Waiting for this to be merged here

**Review by shilman — APPROVED** [review]
@IanVS @tmeasday 

There are some messed up chromatic snapshots (controls, etc.), also failing in my previous Vue2 PR. I'm going to merge this PR and follow up separately. It's too painful to deal with the lack of cached repro template. In the future, we should split up the PRs so that we can merge the repro templates first and then iterates on CI errors with a stable repro.

**Pull request merged**